azurelinuxagent/common/protocol/extensions_config_retriever.py

```diff
@@ -0,0 +1,348 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import os
+import re
+
+from azurelinuxagent.common.future import ustr
+
+import azurelinuxagent.common.conf as conf
+import azurelinuxagent.common.logger as logger
+from azurelinuxagent.common.protocol.goal_state import ExtensionsConfig
+from azurelinuxagent.common.utils import fileutil, restutil
+from azurelinuxagent.common.exception import ProtocolError
+from azurelinuxagent.common.utils.shellutil import run_command
+from azurelinuxagent.common.utils.textutil import safe_shlex_split
+
+INCARNATION_FILE_NAME = "Incarnation"
+SEQUENCE_NUMBER_FILE_NAME = "ArtifactProfileSequenceNumber"
+SVD_SEQNO_FILE_NAME = "SvdSeqNo"
+GOAL_STATE_SOURCE_FILE_NAME = "GoalStateSource"
+VM_ID_FILE_NAME = "VmId"
+
+GOAL_STATE_SOURCE_FABRIC = "Fabric"
+GOAL_STATE_SOURCE_FASTTRACK = "FastTrack"
+
+DMIDECODE_CALL = "dmidecode"
+
+""" 
+GenericExtensionsConfig abstracts whether we pulled the goal state from Fabric or from FastTrack
+consumers should not worry from where the ExtensionsConfig came. They should also have no knowledge
+of sequence numbers or incarnations, which are specific to FastTrack and Fabric respectfully
+"""
+class GenericExtensionsConfig(object):
+    def __init__(self, extensions_config, changed, ext_conf_retriever):
+        self.extensions_config = extensions_config
+        self.changed = changed
+        self._ext_conf_retriever = ext_conf_retriever
+
+        # Copy all properties from extensions_config to make this look like one
```

Comment:
The idea is to keep the logic for determining FT vs Fabric in one place. By localizing the code here, we also make way for other changes in the GA that won't have to worry about FT vs Fabric.

Reply:
I totally get that. I mainly meant from an implementation point of view, why not have `GenericExtensionsConfig` class inherit from the `ExtensionsConfig` class instead of using this (`self.__dict__.update(extensions_config.__dict__)`) to upate the properties. Won't that serve the same purpose?

Reply:
That makes sense and I agree the code is more straightforward, so I derived it from ExtensionsConfig. The only thing is we wind up parsing the xml twice because GenericExtensionsConfig accepts an ExtensionsConfig. If that's a huge issue I can change it, but it would be a bigger change.

Reply:
Thanks for changing it, added another comment regarding this